src/goto-symex/show_program.cpp

```diff
@@ -78,3 +84,243 @@ void show_program(const namespacet &ns, const symex_target_equationt &equation)
       show_step(ns, step, "SHARED_WRITE", count);
   }
 }
+
+bool duplicated_previous_step(const SSA_stept &ssa_step)
+{
+  return !(
+    ssa_step.is_assignment() || ssa_step.is_assert() || ssa_step.is_assume() ||
+    ssa_step.is_constraint() || ssa_step.is_shared_read() ||
+    ssa_step.is_shared_write());
+}
+
+void show_ssa_step_plain(
+  messaget::mstreamt &out,
+  const namespacet &ns,
+  const SSA_stept &ssa_step,
+  const exprt &ssa_expr)
+{
+  const irep_idt &function_id = ssa_step.source.function_id;
+  const std::string ssa_expr_as_string = from_expr(ns, function_id, ssa_expr);
+
+  out << messaget::faint << "// " << ssa_step.source.pc->location_number << " ";
+  out << ssa_step.source.pc->source_location.as_string() << "\n"
+      << messaget::reset;
+  out << ssa_expr_as_string << "\n";
+}
+
+void show_ssa_step_json(
+  std::ostream &out,
+  const namespacet &ns,
+  const SSA_stept &ssa_step,
+  const exprt &ssa_expr,
+  json_arrayt &byte_op_list)
```

Comment:
:pick: As a point of best-practice, I would prefer to see a function's return value used for its result rather than an in/out parameter. For example with this function you could return the `json_objectt` for a step, rather than mutating the `json_arrayt &byte_op_list` parameter. Adding the object to the array would then be done in the calling code.

Reply:
Comment addressed. Pushed changes.
All json_objectt are now returned from functions.